0051-options-local-pragma.rst

```diff
@@ -0,0 +1,106 @@
+Options Local Pragma
+====================
+
+.. proposal-number:: 
+.. ticket-url::
+.. implemented::
+.. highlight:: haskell
+.. header:: This proposal is `discussed at this pull request <https://github.com/ghc-proposals/ghc-proposals/pull/0>`_.
+.. sectnum::
+.. contents::
+
+We propose to add an opportunity to switch GHC-options locally. For instance, this code
+::
+ f = {-# OPTIONS_LOCAL -Wname-shadowing #-}
+   let a = () in
+   let a = 5 in
+     a + a
+get warning ``-Wname-shadowing`` to line
+::
+ let a = 5 in
+
+Motivation
+------------
+
+There are the following reasons to implement such pragma:
+
+1) **Local control of warnings**
+
+To allow control warnings when you confident in your decisions without passing really important warnings.
+
+Warnings are compiler recomendations and often they can be really useful. It allows to avoid stupid mistakes and points to *weak* places in the code. You should follow it. But sometimes you can be confident in your locally code-decision and get some useless warning. It makes tracking warning list more difficult and you can pass really important warning.
+
+ Consider two situations:
+ 
+ a) You want to enable warning locally (and it will be disables in all file).
+ ::
+  f = {-# OPTIONS_LOCAL -Wname-shadowing #-}
+    let a = () in
+    let a = 5 in
+      a + a
+
+ b) You want to disable warning locally (and it will be enabled locally):
+ ::
+  {-# OPTIONS -Wunused-do-bind #-}
```

Comment:
```suggestion
  {-# OPTIONS_GHC -Wunused-do-bind #-}
```

Reply:
I checked `OPTIONS`. Why is using `OPTIONS_GHC` more preferable?

Reply:
> Previous versions of GHC accepted `OPTIONS` rather than `OPTIONS_GHC`, but that is now deprecated.

https://downloads.haskell.org/~ghc/latest/docs/html/users_guide/glasgow_exts.html#options-ghc-pragma